### PR TITLE
Run kyma-integration selectively

### PR DESCRIPTION
**Pull request by Szymongib**
**Description**

Changes proposed in this pull request:

- kyma-integration job runs on PR only if there were some changes to `resources` or `installation` directories
- kyma-integration runs on master branch after commit

As for now with job per component approach, it is not possible to run kyma-integration job only in case all components jobs succeeded.

**Related issue(s)**
#46

### CODE REVIEW

File: prow/config.yaml [hunk 1]
```diff
@@ -37,5 +38,19 @@ presubmits: # runs on PRs
       containers:
       - image: eu.gcr.io/kyma-project/snapshot/test/integration:0.0.1 # created by running `docker build -t <image> .` in the integration-job directory.
 
+postsubmits:
+  kyma-project/kyma:
+  - name: kyma-integration
+    branches:
+    - master
+    context: kyma-integration
+    skip_report: false # from documentation: SkipReport skips commenting and setting status on GitHub.
```
Comment (michal-hudy) [inline prow/config.yaml]:
```suggestion
    skip_report: true # from documentation: SkipReport skips commenting and setting status on GitHub.
    optional: true
```

File: prow-configurations/config.yaml.tpl [hunk 2]
```diff
@@ -36,6 +37,20 @@ presubmits: # runs on PRs
...
       containers:
       - image: eu.gcr.io/kyma-project/snapshot/test/integration:0.0.1 # created by running `docker build -t <image> .` in the integration-job directory.
+            
+postsubmits:
+  {{ .OrganizationOrUser }}/kyma:
+  - name: kyma-integration
+    branches:
+    - master
+    context: kyma-integration
+    skip_report: false # from documentation: SkipReport skips commenting and setting status on GitHub.
```
Comment (michal-hudy) [inline prow-configurations/config.yaml.tpl]:
```suggestion
    skip_report: true # from documentation: SkipReport skips commenting and setting status on GitHub.
    optional: true
```

File: prow/config.yaml (hunk 1 above)
Comment (Szymongib) [reply prow/config.yaml]: There is no `optional` and `skip_report` configurations on postsubmits jobs therefor I have removed it.

File: prow-configurations/config.yaml.tpl (hunk 2 above)
Comment (Szymongib) [reply prow-configurations/config.yaml.tpl]: Same as above, postsubmits do not have `optional` or `skip_report` config

File: prow-configurations/config.yaml.tpl [hunk 3]
```diff
@@ -36,6 +37,19 @@ presubmits: # runs on PRs
     spec:
       containers:
       - image: eu.gcr.io/kyma-project/snapshot/test/integration:0.0.1 # created by running `docker build -t <image> .` in the integration-job directory.
+            
+postsubmits:
+  {{ .OrganizationOrUser }}/kyma:
+  - name: kyma-integration
```
Comment (aszecowka) [inline prow-configurations/config.yaml.tpl]: IMO name and context should be kyma-integration-postsubmit or kyma-integration-master

Comment (Szymongib) [reply prow-configurations/config.yaml.tpl]: Done

**Pull request merged**